src/Payum/Core/CoreGatewayFactory.php

```diff
@@ -59,10 +66,37 @@ public function createConfig(array $config = array())
         $config['twig.env'] = null;
 
         $config->defaults(array(
-            'payum.template.layout' => '@PayumCore/layout.html.twig',
-
-            'payum.http_client' => HttpClientFactory::create(),
+            'httplug.client'=>function(ArrayObject $config) {
+                if (class_exists(HttpGuzzle6Client::class)) {
+                    return new HttpGuzzle6Client(new GuzzleClient());
+                }
+                
+                // TODO add "if else" for other clients provided by httplug.
+
+                if (class_exists(HttpClientDiscovery::class)) {
+                    return HttpClientDiscovery::find();
+                }
+
+                throw new \LogicException('The httpplug.client could not be guessed. Install one of the following packages: php-http/guzzle6-adapter. You can also overwrite the config option with your implementation.');
+            },
+            'httplug.message_factory'=>function(ArrayObject $config) {
+                if (class_exists(GuzzleClient::class)) {
+                    return new GuzzleMessageFactory();
+                }
+
+                // TODO add "if else" for other message factories provided by httplug.
+                
+                if (class_exists(MessageFactoryDiscovery::class)) {
```

Comment:
It would be good to try discovery first and only after fallback to "my solution"